src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -100,7 +100,7 @@ public function validate($value, Constraint $constraint)
 
         if (Email::VALIDATION_MODE_STRICT === $constraint->mode) {
             if (!class_exists('\Egulias\EmailValidator\EmailValidator')) {
-                throw new RuntimeException('Strict email validation requires egulias/email-validator ~1.2|~2.0');
+                throw new LogicException('Strict email validation requires egulias/email-validator ~1.2|~2.0');
```

Comment:
That's a BC break as LogicException does not extend RuntimeException. Not sure it is worth it.

Reply:
True, but nobody can rely on this exception being thrown because a package is missing, don't you think?

Reply:
True, but I don't see why Logic is better than Runtime. We are using Runtime for such exceptions on the Messenger component for instance and Logic on some other components.

Reply:
To me, a logic and exception means the code is broken (composer.json here): it's a programmer mistake. Runtime exception means sope unexpected behavior was observed (e.g. broken connection, etc.)

Reply:
Works for me, but we should fixed this across the board then.

Reply:
Done.